### DISCUSSION TITLE
How to use existing AppArmor profiles?

### DISCUSSION THREAD

**Discussion by realjench** [netblue30/firejail#5888] [Q&A]
Hi.

I'm struggling to make firejail use by default existing apparmor profiles for each application that have one.

I read that using apparmor and firejail may not be a good idea.

I don't want to create specifics profiles for applications, too many to do.

I don't like the idea to use apparmor profile firejail-default, I think security would be worse than using the existing profiles for each apps.

Further more, aa-logprof was broken by firejail-default...

And why is there no any tool like aa-log prof to interactivly correct a firejail profile?

Thanks a lot

**Comment by rusty-snake** [comment]
> I'm struggling to make firejail use by default existing apparmor profiles for each application that have one.

https://github.com/netblue30/firejail/blob/2b34747db534f5db2f577f31d685824b29537098/src/man/firejail-profile.txt#L493-L495

> And why is there no any tool like aa-log prof to interactivly correct a firejail profile?

Because nobody wrote it yet. Or even the technological foundation for it.

**Comment by realjench** [comment]
> > I'm struggling to make firejail use by default existing apparmor profiles for each application that have one.
> 
> https://github.com/netblue30/firejail/blob/2b34747db534f5db2f577f31d685824b29537098/src/man/firejail-profile.txt#L493-L495
> This only can specify one apparmor profile instead of the default firejail-default apparmor profile.

I mean't, that firejail, automaticaly look for an existing profile by himself and use it, is it possible or the only way to do that it to create custom profiles for firejail (to avoid to be erased on next update)?

**OR maybe just rename the apparmor profile to make it used autmatically by firejail would be enough? How to rename it if it's the case?**

Bing is quite funny, he tell me to put for example :
# Enable AppArmor confinement
apparmor /etc/apparmor.d/usr.bin.firefox

in a firejail profile, but apparmor can't take parameter... Would be a so powerfull feature!

Thanks a lot for your quick answer!

And for the tool I understand, by the way I love jailcheck! A tool that apparmor don't have LOL

**Comment by ghost** [comment]
> Bing is quite funny, he tell me to put for example:
apparmor /etc/apparmor.d/usr.bin.firefox
in a firejail profile, but apparmor can't take parameter

Bing is specifying the syntax Firejail uses, not AppArmor's. I agree the man page isn't very clear on the topic. The only profile I could find that uses such a parameter is fdns.profile:
https://github.com/netblue30/firejail/blob/2b34747db534f5db2f577f31d685824b29537098/etc/profile-a-l/fdns.profile#L24

To me that suggests it is using the AA profile I can see under /etc/apparmor.d/usr.bin.fdns. Not the most convenient way to refer to those AA profiles, but that's what we've got.

If you decide to use only AA, there's a setting in firejail.config to disable Firejail's AA functionality:
https://github.com/netblue30/firejail/blob/2b34747db534f5db2f577f31d685824b29537098/etc/firejail.config#L9-L10

**Reply by realjench** [reply]
Thanks again for the time you took to answer!

Finally, I use firecfg to bind another directory for only few applications.

Then a script that modify apparmor firejail-default to include rules specific to an application (or several )then reload the profile before launching the application.

That's convenient enough I find, I like firejail and apparmor, tuning the two is perfect.

Because when you disable firejail's apparmor, you disable apparmor for the application too
